cmake/kokkos_functions.cmake

```diff
@@ -224,12 +224,6 @@ MACRO(kokkos_import_tpl NAME)
     SET(TPL_IMPORTED_NAME Kokkos::${NAME})
   ENDIF()
 
-  # Even though this policy gets set in the top-level CMakeLists.txt,
-  # I have still been getting errors about ROOT variables being ignored
-  # I'm not sure if this is a scope issue - but make sure
-  # the policy is set before we do any find_package calls
-  CMAKE_POLICY(SET CMP0074 NEW)
-
```

Comment:
@masterleinad Try reverting this hunk only.

Reply:
This works for my case. I think the point of this pull request was not to change any behavior, though, right?
It seems that the one in `CMakeLists.txt` doesn't help for my case but I would keep the other ones so that `Kokkos` can be found through `Kokkos_ROOT`.

Reply:
Moving `CMAKE_POLICY(SET CMP0074 NEW)` up in `CMakeLists.txt` to the same place we are dealing with other policies also works (without the one in `cmake/kokkos_functions.cmake` also works and seems preferable.

Reply:
The issue is that we don't call `cmake_minimum_required` early enough